### PR TITLE
Store Products: handle failed requests from the remote site

### PR DESCRIPTION
**Pull request by ryelle**
If the remote site for a store ends up misconfigured (we can't communicate to the API for some reason), there's an error when trying to fetch products. See:

store-errors

This PR fixes the issue by checking that the products returned is not an API error object.

**To test**

- Break your store - I disabled wc-api-dev on a Pressable (not atomic) site
- Load up the dashboard
- View in the console that there are no errors (unless you have debug on, in which case you'll see a message about the API not working)
- The dashboard should "finish" loading, it'll be blank. A future PR could put a better error message here.

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/products/actions.js [hunk 1]
```diff
@@ -157,6 +159,16 @@ export function fetchProducts( siteId, params, successAction, failureAction ) {
 }
 
 export function productsUpdated( siteId, params, products, totalPages, totalProducts ) {
+	// This passed through the API layer successfully, but failed at the remote site.
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/products/actions.js]: Could we add a unit test for this scenario?

Comment (ryelle) [reply client/extensions/woocommerce/state/sites/products/actions.js]: Added :) I put it at the data layer, so we know that the correct action is fired based on the API response.

**Comment by ryelle** [comment]
I've switched to checking that the products response we get is an array, rather than relying on keys in the response (because regardless of what the response is, it's an issue if it's not an array).

Unfortunately (or fortunately?), the `Products API Error:  Error: Unknown Token` isn't actually an error– I mean, it's an issue that we can't get to their API, but it's clear that the underlying issue is that the site is still a wp.com site. You can trigger it by going to `/store/:site` for any wp.com site. It's actually the data-layer doing what I told it to — [on an error from the API, it should warn about it](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/data-layer/products/index.js#L41).

We should probably focus on not trying to request this data unless we know it's an atomic site.

**Comment by timmyc** [comment]
I tried to reproduce the error on a wpsandbox.me site with `wc-api-dev` disabled and was unable to do so - otherwise this PR is looking good for me, just wish I had a way to test...

**Pull request merged**